tests/ConcurrencyTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
```

Comment:
It would be testing promises, Guzzle bindings with the SDK, SDK clients, et. al. At minimum, I think we need integration tests for the bindings (I can't remember if those exist somewhere else offhand). Since we ship the SDK with Guzzle bindings, I think we should have an integration test that makes sure those bindings work correctly. I think integration tests for this are just a really easy way to make sure something isn't super broken before a release. It's nice just to make sure that the bindings are correct I think.